mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/options/MongoDBCollection.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.options;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.hibernate.ogm.datastore.mongodb.options.impl.MongoDBCollectionConverter;
+import org.hibernate.ogm.options.spi.MappingOption;
+
+/**
+ * MongoDB specific definitions for a collection.
+ *
+ * @author Guillaume Smet
+ */
+@Target({ TYPE, METHOD, FIELD })
+@Retention(RUNTIME)
+@MappingOption(MongoDBCollectionConverter.class)
+public @interface MongoDBCollection {
```

Comment:
These options as described by https://docs.mongodb.com/manual/reference/method/db.createCollection/ are all about "DDL" generation tuning and very very specific to MongoDB. Most are actually specific to new versions of MongoDB.

For the sake of simplicity, I would even go as far as offering a full bypass.

    @MongoDBCollectionOptions(name="customerCollection", options="{ capped : true, size : 5242880, max : 5000, storageEngine: { wiredTiger: { configString: \"<option>=<setting>\" } } }")

While we should try and align on runtime things like the target structure and maybe even whether a property is indexed or not, I don't think the DDL options are a worth it. To be honest even Index will be hard. For example Infinispan indexes are infinitely complex as we use the full power of Hibernate Search.